addons/im_livechat/models/discuss_channel.py

```diff
@@ -415,7 +415,7 @@ def _message_post_after_hook(self, message, msg_vals):
                 .browse(self.env.context.get("selected_answer_id"))
                 .exists()
             )
-            if selected_answer in self.chatbot_current_step_id.answer_ids:
+            if selected_answer and selected_answer in self.chatbot_current_step_id.answer_ids:
```

Comment:
I'd rather not enter the condition block. Why not update the condition above to:

```py
        if (
            selected_answer_id := self.env.context.get("selected_answer_id")
            and self.chabtot_current_step_id
        ):
            selected_answer = self.env["chatbot.script.answer"].browse(selected_answer_id).exists()
```

Reply:
I understand, thanks. However, I think skipping the creation of the `chatbot.message` only for notification messages isn't enough. I feel we should skip creating chatbot messages as soon as an agent is found (when `livechat_agent_history_ids` isn't empty).